src/lib/matcher.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * This module provides the basic type machinery to build pipeline
+ * translators efficiently and safely.
+ *
+ * It exposes 2 main building blocks:
+ * - `StepMatcher`: a type mapping each available pipeline steps to a
+ *   translation function
+ * - `matchStep`: a function accepting a `StepMatcher` and that return a function
+ *   transforming a pipeline step to an arbiratry output.
+ *
+ * This system is supposed to be typesafe, which means that if you add a
+ * new type of `PipelineStep`, the typescript compiler is expected to fail
+ * if your matcher implementation doesn't provide a mapping for this new kind
+ * of step.
```

Comment:
is this comment still true ? Reading the doc I understood that the it would automatically detect a new serie of unsuported steps in translators in this case.

Reply:
The `StepMatcher` is still the backbone the system relies on.